src/OSPSuite.Core/Domain/Services/ConcurrencyManager.cs

```diff
@@ -7,6 +7,22 @@
 
 namespace OSPSuite.Core.Domain.Services
 {
+   public class ConcurrencyManagerResult<TResult>
+   {
+      public string Id { get; }
+      public bool Succeeded { get; }
+      public string ErrorMessage { get; }
+      public TResult Result { get; }
+
+      public ConcurrencyManagerResult(string id, bool succeeded, TResult result, string error)
+      {
+         Id = id;
+         Succeeded = succeeded;
+         ErrorMessage = error;
```

Comment:
Either call the parameter errorMessage or the Member Error. No reason to have different names here.